pingouin/multivariate.py

```diff
@@ -254,3 +254,114 @@ def multivariate_ttest(X, Y=None, paired=False):
     stats = {'T2': t2, 'F': fval, 'df1': df1, 'df2': df2, 'pval': pval}
     stats = pd.DataFrame(stats, index=['hotelling'])
     return _postprocess_dataframe(stats)
+
+
+def box_m(data, dvs=None, group=None, alpha=.001):
+    """Test equality of covariance matrices.
+
+    Parameters
+    ----------
+    data : :py:class:`pandas.DataFrame`,
+    should be a long-format pandas dataframe.
+    dvs : str
+        Variables of covariance matrix (only when ``data``
+        is a long-format dataframe).
+    group : str
+        Grouping variable (only when ``data`` is a long-format dataframe).
+    alpha : float
+        Significance level.
+
+    Returns
+    -------
+    stats : :py:class:`pandas.DataFrame`
+
+        * ``'Chi2'``: Test statistic
+        * ``'pval'``: p-value
+        * ``'df'``: The Chi-Square statistic's degree of freedom
+        * ``'equal_cov'``: True if ``data`` has equal covariance
+
+    Notes
+    -----
+    This function does not handle missing values. Please
+    handle them (e.g. remove or impute these values)
+    prior to perform the Box's M test.
+    Pooled sample covariance matrix :math:`S_{\\text{pl}}`
+    is calculated as
+    .. math::
+
+        S_{\\text{pl}} = \\frac{\\sum_{i=1}^k(n_i-1)
+        \\textbf{S}_i}{\\sum_{i=1}^k(n_i-1)},
+    where :math:`n_i` and :math:`S_i` are the sample size and covariance matrix
+    of the :math:`i^{th}` sample, :math:`k` is
+    the number of independent samples.
+    More mathematical expressions can be found in [1]
+
+    References
+    ----------
+    .. [1] Rencher, A. C. (2003). Methods of multivariate analysis (Vol. 492).
+    John Wiley & Sons.
```

Comment:
Could you also add the reference for using 0.001 as the default significance value? Thanks